torch/utils/_cxx_pytree.py

```diff
@@ -187,20 +203,25 @@ def register_pytree_node(
             )
         )
     """
-    from ._pytree import _register_pytree_node
-
-    _register_pytree_node(
-        cls,
-        flatten_func,
-        unflatten_func,
-    )
+    if _register_python_pytree_node:
+        from ._pytree import _register_pytree_node
+
+        _register_pytree_node(
+            cls,
+            flatten_func,
+            unflatten_func,
+            to_dumpable_context=to_dumpable_context,
+            from_dumpable_context=from_dumpable_context,
+            _register_cxx_pytree_node=False,
+        )
 
-    optree.register_pytree_node(
-        cls,
-        flatten_func,
-        _reverse_args(unflatten_func),
-        namespace=namespace,
-    )
+    if not optree.is_structseq_class(cls):
+        optree.register_pytree_node(
+            cls,
+            flatten_func,
+            _reverse_args(unflatten_func),
+            namespace=namespace,
+        )
```

Comment:
Special case for PyStructSequence types in `torch.return_types`. Will remove this after we make Python pytree also out-of-box support PyStructSequence types.